### PR TITLE
configure implementations with service ids

### LINKED ISSUE #9: improve bundle configuration
requirement: using a custom implementation (event store, read model, saga) should be configurable in the bundle by specifying a service id, e.g.:

``` yaml
broadway:
    event_store:
            type: service
            id: my_acme.event_store
```

as proposed in
* https://github.com/broadway/broadway/issues/227
* https://github.com/broadway/broadway/pull/274
* https://github.com/broadway/broadway-bundle/pull/16

or

``` yaml
broadway:
    event_store: my_acme.event_store # default "broadway.event_store.in_memory"
```

see POC in https://github.com/broadway/broadway-bundle/pull/18/commits/1d89930bfbdce2f75dbd049e52960f04d6520536

another solution is to tag your custom implementation and have a compile pass register them.

### PR DESCRIPTION
**Pull request by othillo**
- [x] configure event store with service id
- [x] configure read model repository factory with service id
- [x] configure saga state repository with service id

- [x] merge https://github.com/broadway/broadway/pull/295
- [x] merge https://github.com/broadway/broadway/pull/294
- [x] merge https://github.com/broadway/event-store-dbal/pull/1
- [x] merge https://github.com/broadway/read-model-elasticsearch/pull/1
- [x] document how to use Doctrine DBAL
- [x] document how to use Elasticsearch
- [x] document how to use MongoDB
- [x] update broadway/broadway-demo
- [ ] tag new release

- [x] decide on usage of doctrine-bundle by the commands: https://github.com/broadway/broadway-bundle/issues/29

fixes https://github.com/broadway/broadway-bundle/issues/9

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -31,6 +31,38 @@ Once enabled the bundle will expose several services, such as:
...
+Broadway provides a persisting event store implementation using `doctrine/dbal`
+in [broadway/event-store-dbal](https://github.com/broadway/event-store-dbal).
+
+This can be installed using composer:
+
+```
+$ composer require broadway/event-store-dbal
+```
+
+You need to configure this event store in you application:
```
Comment (wjzijderveld) [inline README.md]: `You will need to configure an event store in your application's service definition`?

File: test/AppKernel.php [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+    public function registerBundles()
+    {
+        return [
+            new BroadwayBundle(),
+        ];
+    }
+
+    public function registerContainerConfiguration(LoaderInterface $loader)
+    {
+        // TODO: Implement registerContainerConfiguration() method.
```
Comment (wjzijderveld) [inline test/AppKernel.php]: Remove todo (or actually do it 😋)

File: README.md [hunk 3]
```diff
@@ -31,6 +31,38 @@ Once enabled the bundle will expose several services, such as:
...
+    <argument>events</argument>
+    <argument>false</argument>
+    <argument type="service" id="broadway.uuid.converter" />
+</service>
+```
+
+And tell the Broadway bundle to use it:
+
+```yaml
+broadway:
```
Comment (wjzijderveld) [inline README.md]: Add `# config.yml` on the first line to provide some context

Comment (wjzijderveld) [reply README.md]: This goes for all configuration code-blocks.

For the service definitions I think it's already obvious enough, but it doesn't hurt adding `<!-- services.xml -->` there as well.

**Review by wjzijderveld — CHANGES_REQUESTED** [review]

**Comment by asm89** [comment]
@wjzijderveld @robinvdvleuten @ricbra Any objections? If none I'll merge this in the next couple of days.

**Comment by robinvdvleuten** [comment]
@asm89 looks very good to me! Nice work @othillo 👍

**Review by wjzijderveld — APPROVED** [review]

**Comment by wjzijderveld** [comment]
Thanks @othillo !

**Comment by othillo** [comment]
@wjzijderveld time to tag 1.0? or just 0.3?

**Pull request merged**